modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -641,7 +641,23 @@ def to_pandas(cls, partitions):
         pandas.DataFrame
             A pandas DataFrame
         """
-        retrieved_objects = [[obj.to_pandas() for obj in part] for part in partitions]
+        retrieved_objects = cls.get_objects_from_partitions(partitions.flatten())
+        if all(
+            map(
+                lambda obj: isinstance(obj, (pandas.DataFrame, pandas.Series)),
+                retrieved_objects,
+            )
+        ):
+            old_shape = partitions.shape
```

Comment:
would it make sense to unpack `partitions.shape` into `width, height = partitions.shape` instead of referring to them as `old_shape[0]` etc? (maybe it should be `height, width =`, I didn't think it through properly)